omocha-domain/src/main/java/org/omocha/domain/auction/Category.java

```diff
@@ -0,0 +1,77 @@
+package org.omocha.domain.auction;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+import jakarta.persistence.CascadeType;
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.JoinColumn;
+import jakarta.persistence.ManyToOne;
+import jakarta.persistence.OneToMany;
+import jakarta.persistence.Table;
+import lombok.AccessLevel;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+
+@Getter
+@Entity
+@Table(name = "category")
+@NoArgsConstructor(access = AccessLevel.PROTECTED)
+public class Category {
+
+	@Id
+	@GeneratedValue(strategy = GenerationType.IDENTITY)
+	private Long categoryId;
+
+	private String name;
+
+	@ManyToOne
+	@JoinColumn(name = "parent_id")
+	private Category parent;
+
+	@OneToMany(mappedBy = "parent", cascade = CascadeType.ALL, orphanRemoval = true)
+	private List<Category> subCategories = new ArrayList<>();
+
+	@OneToMany(mappedBy = "category")
+	private List<AuctionCategory> auctionCategories = new ArrayList<>();
+
+	@Builder
+	public Category(String name, Category parent) {
+		this.name = name;
+		this.setParent(parent);
+	}
+
+	public void setParent(Category parent) {
+		if (this.parent != null) {
+			this.parent.getSubCategories().remove(this);
+		}
+		this.parent = parent;
+		if (parent != null && !parent.getSubCategories().contains(this)) {
+			parent.getSubCategories().add(this);
+		}
```

Comment:
parent를 추가하고 parent가 null이 아니면서 parent가 자신을 포함하고 있지 않다면 추가하는 코드로 보여지는데
이렇게 처리를 하면 dirty check로 parent까지 한번에 update 되는게 맞을까요?

```suggestion
		if (parent != null && !parent.getSubCategories().contains(this)) {
			this.parent = parent;
			parent.getSubCategories().add(this);
		}
```
또 이렇게 처리를 하는게 맞지 않을까 싶긴 한데...

Reply:
CategoryServiceImpl 에 있는 addCategory와 같이 보시면 될 것 같아요.

parentCategory를 저장할때 null이면 parent에 null 값이 저장되고 subcategory에는 저장이 안됩니다. 그래서 this.parent=parent가 밖에 있어야함.

그리고 만약에 하위 계층이 생겨서 parent가 존재를 한다면 부모의 subCategory에 자식을 담아주는 로직입니다.

Reply:
아하 이해했습니다!